src/Processor/Braintree/Mapper/Subscription/ModifierMapper.php

```diff
@@ -102,6 +102,43 @@ private function withRemovedPreviousPlanDefaults(array &$request, array $default
         return $request;
     }
 
+    /**
+     * Determines if a modifier should be added or modified on an existing Subscription based off of the Plan defaults & which modifiers are already on the Subscription.
+     */
+    private function isModifierAddedOnExistingSubscription(bool $isDefaultModifier, Subscription $subscription, ModifierType $modifierType, Modifier $modifier): bool
+    {
+        $hasFnName = $modifierType->equals(ModifierType::addOn()) ? 'hasAddOn' : 'hasDiscount';
```

Comment:
Does this all jive well with the static analysis? This passes linting, static analysis, and tests?

Reply:
Yes. I think adding the `@var` types a little further down where the functions are used is what probably does the trick. There's a screenshot in the PR description that shows the tests and static analysis passing.

I can flip this back to more similar to how it was previously if you prefer. We'll just have some duplicated code.

Reply:
I don't care if the static analysis doesn't care. I bless this 🕊️